plugins/inputs/solrmetrics/solrmetrics.go

```diff
@@ -0,0 +1,413 @@
+	mTime := time.Now()
+	series := make(map[string]Serie)
+
+	// Create series for standard groups
+	for _, group := range []string{"jvm", "jetty", "node"} {
+		series[group] = Serie{mesName: group, fields: make(Fields), tags: make(Tags), timestamp: mTime}
+		copyStrMaps(addTags, series[group].tags)
+	}
+
+	// Parce metrics by a group
+	for k, v := range metrics {
+		// standard groups
+		for _, group := range []string{"jvm", "jetty", "node"} {
+			if strings.Contains(k, "solr"+delimeter+group) {
+				newSerie := strings.Replace(k, "solr."+group+":", "", 1)
+				series[group].fields[newSerie] = v
+			}
+		}
+		// Core group
+		// Each core of any collection should be present as a separate serie with uniq tags
+		if strings.Contains(k, "solr.core") {
+			newSerie := strings.Replace(k, "solr.core"+delimeter, "", 1)
+			// Decompose a name of a metric
+			colTags, err := getCollectionTags(newSerie)
+			if err != nil || len(colTags) != 4 {
+				acc.AddError(err)
+				continue
+			}
+			collection, shard, replica, metricName := colTags[0], colTags[1], colTags[2], colTags[3]
+			newSerie = collection + shard + replica
+			// Check and create an unique serie if it doesn't exist
+			if _, ok := series[newSerie]; !ok {
+				series[newSerie] = Serie{mesName: "core", fields: make(Fields), tags: make(Tags), timestamp: mTime}
+				// Add requeremnted tags
+				copyStrMaps(addTags, series[newSerie].tags)
+				series[newSerie].tags["collection"] = collection
+				series[newSerie].tags["shard"] = shard
+				series[newSerie].tags["replica"] = replica
+			}
+			series[newSerie].fields[metricName] = v
+		}
+	}
+	res = []Serie{}
+	for _, v := range series {
+		// Add a serie, if fileds contain metrics
+		if len(v.fields) > 0 {
+			res = append(res, v)
+		}
+	}
+	return res
+}
+
+// SampleConfig returns sample configuration for this plugin.
+func (s *Solr) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the plugin description.
+func (s *Solr) Description() string {
+	return "Read stats from Solr servers and cores using Metrics API"
+}
+
+// NewSolr return a new instance of Solr
+func NewSolr() *Solr {
+	return &Solr{}
+}
+
+// Provide metrics urls
+func (s *Solr) metricsURLs(acc telegraf.Accumulator) ([]url.URL, error) {
+	if len(s.Prefixes) == 0 && len(s.Keys) == 0 {
+		return nil, fmt.Errorf("'keys' and 'prefix' are empty")
+	}
+	URLs := make([]url.URL, 0, len(s.Servers)*2)
+	for _, server := range s.Servers {
+		URLPrs := make(map[string]url.Values, 0)
+		if len(s.Keys) > 0 {
+			URLPrs["keys"] = url.Values{}
+		}
+		for _, key := range s.Keys {
+			URLPrs["keys"].Add("key", key)
+		}
+		if len(s.Prefixes) > 0 {
+			URLPrs["prefixes"] = url.Values{}
+			URLPrs["prefixes"].Set("prefix", strings.Join(s.Prefixes, ","))
+		}
+		// URL for a server
+		strURL := fmt.Sprintf("%s/solr%s", server, metricPath)
+		serverURL, err := url.Parse(strURL)
+		if err != nil {
+			acc.AddError(err)
+			continue
```

Comment:
Please simply return the error and stop the initialization altogether. Make sure to fail early and loudly!